packages/core/database/lib/entity-manager.js

```diff
@@ -615,40 +627,117 @@ const createEntityManager = (db) => {
           const { joinTable } = attribute;
           const { joinColumn, inverseJoinColumn } = joinTable;
 
-          // clear previous associations in the joinTable
-          await this.createQueryBuilder(joinTable.name)
-            .delete()
-            .where({ [joinColumn.name]: id })
-            .where(joinTable.on || {})
-            .execute();
+          const onlyDeleteRelation = _.isNull(data[attributeName]);
+          let insert;
 
-          if (
-            isBidirectional(attribute) &&
-            ['oneToOne', 'oneToMany'].includes(attribute.relation)
-          ) {
+          if (onlyDeleteRelation) {
+            // clear previous associations in the joinTable
             await this.createQueryBuilder(joinTable.name)
               .delete()
-              .where({ [inverseJoinColumn.name]: toIds(data[attributeName]) })
+              .where({ [joinColumn.name]: id })
               .where(joinTable.on || {})
               .execute();
-          }
+          } else {
+            const cleanRelationData = toAssocs(data[attributeName]);
+            const connect = cleanRelationData.connect || [];
+            const disconnect = cleanRelationData.disconnect || [];
+            const isPartialUpdate = connect.length || disconnect.length;
+
+            if (isPartialUpdate) {
+              const idsToRemove = _.map('id', connect.concat(disconnect));
+              const existingRelsToDelete = await this.createQueryBuilder(joinTable.name)
+                .select([inverseJoinColumn.name, 'order'])
+                .where({
+                  [joinColumn.name]: id,
+                  [inverseJoinColumn.name]: { $in: idsToRemove },
+                })
+                .where(joinTable.on || {})
+                .execute();
 
-          if (!_.isNull(data[attributeName])) {
-            const insert = toAssocs(data[attributeName]).map((data) => {
-              return {
-                [joinColumn.name]: id,
-                [inverseJoinColumn.name]: data.id,
-                ...(joinTable.on || {}),
-                ...(data.__pivot || {}),
-              };
-            });
+              await this.createQueryBuilder(joinTable.name)
+                .delete()
+                .where({
+                  [joinColumn.name]: id,
+                  [inverseJoinColumn.name]: { $in: idsToRemove },
+                })
+                .where(joinTable.on || {})
+                .execute();
 
-            // if there is nothing to insert
-            if (insert.length === 0) {
-              continue;
-            }
+              for (const relToDelete of existingRelsToDelete) {
+                await this.createQueryBuilder(joinTable.name)
+                  .update({ order: db.getConnection().raw('?? - 1', 'order') })
+                  .where({
+                    [joinColumn.name]: id,
+                    order: { $gt: relToDelete.order },
```

Comment:
do we actually need an order column for this first iteration?
We could just have no order and add the new ones at the end.